contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+                     * _checkLiquidityWithinExpectedShare
+                     *
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: wethBalance,
+                        amount1Desired: oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _updateUnderlyingAssets();
+            emit LiquidityAdded(
+                wethBalance, // wethAmountDesired
+                oethbRequired, // oethbAmountDesired
+                wethAmountSupplied, // wethAmountSupplied
+                oethbAmountSupplied, // oethbAmountSupplied
+                tokenId, // tokenId
+                underlyingAssets
+            );
+        }
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the liquidity in the pool is withing the expected WETH to OETHb ratio
+     */
+    function _checkLiquidityWithinExpectedShare() internal {
+        (
+            uint256 wethPositionBalance,
+            uint256 oethbPositionBalance
+        ) = getPositionPrincipal();
+        require(
+            wethPositionBalance + oethbPositionBalance > 0,
+            "No liquidity in position"
+        );
+
+        uint160 currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(
+            currentPrice >= sqrtRatioX96Tick0 &&
+                currentPrice <= sqrtRatioX96Tick1,
+            "Not in expected tick range"
+        );
+
+        uint256 currentWethShare = 0;
+        if (wethPositionBalance != 0) {
+            currentWethShare = wethPositionBalance.divPrecisely(
+                wethPositionBalance + oethbPositionBalance
+            );
+        }
+
+        uint256 wethDiff = Math.max(poolWethShare, currentWethShare) -
+            Math.min(poolWethShare, currentWethShare);
+
+        if (wethDiff < poolWethShareVarianceAllowed) {
+            emit PoolRebalanced(
+                currentWethShare,
+                poolWethShare,
+                wethPositionBalance,
+                oethbPositionBalance
+            );
+        } else {
+            revert PoolRebalanceOutOfBounds(
+                currentWethShare,
+                poolWethShare,
+                wethPositionBalance,
+                oethbPositionBalance
+            );
+        }
+    }
+
+    /**
+     * Burns any lingering OETHb tokens still remaining on the contract
+     */
+    function _burnOethbOnTheContract() internal {
+        uint256 oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethbBalance);
```

Comment:
nit: `oethbBalance > 0` check here may be? However, neither `IVault.burnForStrategy` nor `OETHb.burn` revert if `amount == 0`, so it's fine

Reply:
good thinking. Leaving it as is it for now